src/hotspot/share/gc/shared/gcArguments.cpp

```diff
@@ -126,7 +129,12 @@ void GCArguments::initialize_heap_flags_and_sizes() {
       vm_exit_during_initialization("Incompatible minimum and maximum heap sizes specified");
     }
   }
-
+  #if defined(LINUX)
+  if (OSContainer::is_containerized() && (OSContainer::memory_and_swap_limit_in_bytes() > 0) &&
+      FLAG_IS_CMDLINE(InitialHeapSize) && (InitialHeapSize >= (julong) OSContainer::memory_and_swap_limit_in_bytes())) {
+    vm_exit_during_initialization("Initial heap size set to a larger value than the container memory & swap limit");
+  }
+  #endif
```

Comment:
`OSContainer::memory_and_swap_limit_in_bytes()` is an expensive operation. It opens files and read their contents.  The returned value should be stored in a local to avoid calling this function twice.